src/Parallel/GlobalCache.hpp

```diff
@@ -100,8 +87,120 @@ using get_component_if_mocked = tmpl::front<tmpl::type_from<tmpl::conditional_t<
+  // gsl::not_null pointer to the object named by the GlobalCacheTag,
+  // and then the contents of 'args' as subsequent arguments.
+  template <typename GlobalCacheTag, typename Function, typename... Args>
+  void mutate(const std::tuple<Args...>& args) noexcept;
+
+  // Not an entry method.
+  template <typename GlobalCacheTag, typename Function>
+  bool mutable_cache_item_is_ready(const Function& function) noexcept;
+
+ private:
+  tuples::tagged_tuple_from_typelist<
+      get_mutable_global_cache_tag_storage<Metavariables>>
+      mutable_global_cache_{};
+};
+
+template <typename Metavariables>
+template <typename GlobalCacheTag>
+auto MutableGlobalCache<Metavariables>::get() const noexcept
+    -> const GlobalCache_detail::type_for_get<GlobalCacheTag, Metavariables>& {
+  using tag =
+      MutableCacheTag<tmpl::front<GlobalCache_detail::get_list_of_matching_tags<
+          GlobalCacheTag, Metavariables>>>;
+  if constexpr (tt::is_a_v<std::unique_ptr, typename tag::tag::type>) {
+    return *(std::get<0>(tuples::get<tag>(mutable_global_cache_)).get());
+  } else {
+    return std::get<0>(tuples::get<tag>(mutable_global_cache_));
+  }
+}
+
+template <typename Metavariables>
+template <typename GlobalCacheTag, typename Function>
+bool MutableGlobalCache<Metavariables>::mutable_cache_item_is_ready(
+    const Function& function) noexcept {
+  using tag = MutableCacheTag<
+      tmpl::front<GlobalCache_detail::get_list_of_matching_mutable_tags<
+          GlobalCacheTag, Metavariables>>>;
+  boost::optional<CkCallback> optional_callback{};
+  if constexpr (tt::is_a_v<std::unique_ptr, typename tag::tag::type>) {
+    optional_callback =
+        function(*(std::get<0>(tuples::get<tag>(mutable_global_cache_)).get()));
+  } else {
+    optional_callback =
+        function(std::get<0>(tuples::get<tag>(mutable_global_cache_)));
+  }
+  if (optional_callback) {
+    std::get<1>(tuples::get<tag>(mutable_global_cache_))
+        .push_back(optional_callback.get());
+    return false;
+  } else {
+    return true;
+  }
+}
+
+template <typename Metavariables>
+template <typename GlobalCacheTag, typename Function, typename... Args>
+void MutableGlobalCache<Metavariables>::mutate(
+    const std::tuple<Args...>& args) noexcept {
+  (void)Parallel::charmxx::RegisterMutableGlobalCacheMutate<
+      Metavariables, GlobalCacheTag, Function, Args...>::registrar;
+  using tag = MutableCacheTag<
+      tmpl::front<GlobalCache_detail::get_list_of_matching_mutable_tags<
+          GlobalCacheTag, Metavariables>>>;
+
+  // Get the callbacks to call, and at the same time
+  // reset the list of stored callbacks to an empty vector.
+  std::vector<CkCallback> callbacks_to_call{};
+  std::swap(callbacks_to_call,
+            std::get<1>(tuples::get<tag>(mutable_global_cache_)));
+
+  // Do the mutate.
+  std::apply(Function::apply,
+             std::tuple_cat(std::forward_as_tuple(make_not_null(&std::get<0>(
```

Comment:
Because mutate is an entry method, it is easier for charm registration purposes and ci files to pass a tuple.

However, I can change the free function Parallel::mutate (the one that users will call) to take a variadic parameter list.